server/src/main/java/org/elasticsearch/index/mapper/BinaryFieldMapper.java

```diff
@@ -65,16 +67,23 @@
     public static class Builder extends FieldMapper.Builder<Builder> {
 
         public Builder(String name) {
-            super(name, Defaults.FIELD_TYPE, Defaults.FIELD_TYPE);
+            super(name, Defaults.FIELD_TYPE);
             builder = this;
         }
 
         @Override
         public BinaryFieldMapper build(BuilderContext context) {
-            setupFieldType(context);
-            return new BinaryFieldMapper(name, fieldType, defaultFieldType,
+            return new BinaryFieldMapper(name, fieldType, new BinaryFieldType(buildFullName(context), hasDocValues, meta),
                     context.indexSettings(), multiFieldsBuilder.build(this, context), copyTo);
         }
+
+        @Override
+        public Builder index(boolean index) {
+            if (index) {
+                throw new MapperParsingException("Binary field [" + name() + "] cannot be indexed");
```

Comment:
is this a new error that gets returned compared to before?

Reply:
It would have previously been caught by the rather byzantine logic in the base class.